CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+cmake_minimum_required (VERSION 3.13.1)
+
+project(UVAtlas)
+
+if (NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "Debug")
+endif()
+
+set(CMAKE_DEBUG_POSTFIX d)
+
+if (WIN32)
+    #Configure compile flags to match VS sln
+    set(CXX_COMMON_FLAGS "/MP /fp:fast /sdl /permissive- /Zc:__cplusplus")
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${CXX_COMMON_FLAGS}")
+    set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} ${CXX_COMMON_FLAGS}")
+    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${CXX_COMMON_FLAGS}")
+
+    #    UVAtlas is too slow in debug so use optimization flags to speed up
+    string(REPLACE "/Od" "/O2" CMAKE_C_FLAGS_DEBUG ${CMAKE_C_FLAGS_DEBUG})
+    string(REPLACE "/Ob0" "" CMAKE_C_FLAGS_DEBUG ${CMAKE_C_FLAGS_DEBUG})
+    string(REPLACE "/RTC1" "" CMAKE_C_FLAGS_DEBUG ${CMAKE_C_FLAGS_DEBUG})
+    string(REPLACE "/Od" "/O2" CMAKE_CXX_FLAGS_DEBUG ${CMAKE_CXX_FLAGS_DEBUG})
+    string(REPLACE "/Ob0" "" CMAKE_CXX_FLAGS_DEBUG ${CMAKE_CXX_FLAGS_DEBUG})
+    string(REPLACE "/RTC1" "" CMAKE_CXX_FLAGS_DEBUG ${CMAKE_CXX_FLAGS_DEBUG})
+elseif(UNIX)
+    set(CXX_COMMON_FLAGS "-std=c++11 -fPIC")
```

Comment:
is it possible to only do this for this project, general `set` makes me nervous

Reply:
Not sure how to make cmake do this. Any pointers? FWIW while I'm updating dependencies I was planning on putting a note in there that the cmake isn't verified to work adding these as submodules. I can add that note to the readme too.

Reply:
you can set it for a specific target

Reply:
So I was trying to set those flags up for a specific target but it doesn't seem to work. I think the issue is that for VS cmake has default compiler flags. Then when you use any of the target specific commands the new flags just get appended to the cl command and get ignored. It seems like the common solution is to use sub-projects and set the `CMAKE_CXX_FLAGS...` from there. They then don't set the global values but still apply to targets within in this part of the cmake tree.